hydrus/data/crud.py

```diff
@@ -695,6 +695,74 @@ def delete_single(type_: str, session: scoped_session) -> None:
     return delete(instance.id, type_, session=session)
 
 
+def insert_modification_record(method: str, resource_url: str,
+                               session: scoped_session) -> str:
+    """
+    Insert a modification record into the database.
+    :param method: HTTP method type of related operation.
+    :param resource_url: URL of resource modified.
+    :param session: sqlalchemy session.
+    :return: ID of new modification record.
+    """
+    modification = Modification(method=method, resource_url=resource_url)
+    session.add(modification)
+    record_count = session.query(Modification).count()
+    if record_count > 1000:
+        oldest_record = session.query(Modification).order_by(Modification.timestamp.asc()).first()
+        session.delete(oldest_record)
+    session.commit()
+    return modification.job_id
+
+
+def get_last_modification_job_id(session: scoped_session) -> str:
+    """
+    Get job id of most recent modification record stored in the db.
+    :param session: sqlalchemy session
+    :return: job id of recent modification.
+    """
+    last_modification = session.query(Modification).order_by(Modification.timestamp.desc()).first()
+    if last_modification is None:
+        last_job_id = ""
+    else:
+        last_job_id = last_modification.job_id
+    return last_job_id
+
+
+def get_modification_table_diff(session: scoped_session,
+                                agent_job_id: str = None) -> List[Dict[str, Any]]:
+    """
+    Get modification table difference.
+    :param session: sqlalchemy session.
+    :param agent_job_id: Job id from the client.
+    :return: List of all modifications done after job with job_id = agent_job_id.
+    """
+    # When client provides a job id but server does not have corresponding record
+    # meaning that client is too outdated.
+    if agent_job_id is not None:
+        try:
+            record_for_agent_job_id = session.query(Modification).filter(
+                Modification.job_id == agent_job_id).one()

```

Comment:
This never gets returned. What's the use?

Reply:
Shouldn't this try to return every element starting from the job_id agent sent? @Guttz @vddesai1871

Reply:
It was for checking whether the client is too outdated or not(if the client tries to get all modification after some `agent_job_id`=`x` and if `x` is not in hydrus DB then it means the client is too outdated).
But yeah it was too complicated, I have refactored it to increase readability.